app/views/project/editProject.js

```diff
@@ -57,11 +58,47 @@ let editProject = (cursor) => {
       })
 }
 
+let DeleteProjectDialog = component('DeleteProjectDialog', (cursor) => {
+  
+  let yesCallback = () => {
+    let editCursor = cursor.cursor('editProject')
+    let projectCursor = editCursor.cursor('project')
+    let project = projectCursor.toJS()
+    let qualifiedProjectId = `${project.owner}/${project.projectId}`
+    
+    editCursor.set('isWaiting', true)
+    
+    ajax.delete(`/api/projects/${project.owner}/${project.projectId}`)
+      .then(() => {
+        logger.debug(`Project successfully deleted '${qualifiedProjectId}'`)
+        router.goTo('/')
+      }, (error) => {
+        logger.warn(`Failed to delete project '${qualifiedProjectId}': ${error}`)
+      })
+      .finally(() => {
+        cursor.set('showDeleteProjectDialog', false)
```

Comment:
I think it would be better to update all of the cursor state at once, as this may be buggy due to cursors being immutable.